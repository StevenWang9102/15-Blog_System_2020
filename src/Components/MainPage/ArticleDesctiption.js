import React from "react";
import { Link } from "react-router-dom";

export const ArticleDesctiption = props => {
  return (
    <Link
      className='nav-link preview-link article-detail'
      onClick={() => props.setLoading("LOADING")}
      to={"/article-detail/" + props.article.slug}>
      <h1>{props.article.title}</h1>
      <p>{props.article.description}</p>
      <span>Read more...</span>
    </Link>
  );
};
