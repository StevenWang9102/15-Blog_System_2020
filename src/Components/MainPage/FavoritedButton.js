import React from "react";
import { useState } from "react";
import { createUseStyles } from "react-jss";
import PropTypes from "prop-types";

const useStyles = createUseStyles({
  myButton: {
    textAlign: "left",
    marginLeft: "10px",
    "&:hover": {
      color: "white",
      fontWeight: "500",
      textAlign: "center",
      backgroundColor: " #5cb85c94"
    },
    "&:focus": {
      color: "#5cb85c",
      fontWeight: "500",
      textAlign: "center",
      backgroundColor: "#5cb85c11"
    }
  }
});

export const FavoritedButton = props => {
  const [favoritesCountPlus, setFavoritesCountPlus] = useState(0);
  const Button = ({ children }) => {
    const classes = useStyles();
    return (
      <button
        type='button'
        className={`btn btn-outline-primary btn-sm pull-xs-right ${classes.myButton}`}
        onClick={() => {
          
          // Switch method between "POST" and "DELETE"
          if (props.userInformation.token) {
            const tempMethod = { ...props.httpMethod };

            if (tempMethod[props.article.slug] === "POST") {
              tempMethod[props.article.slug] = "DELETE";
            } else {
              tempMethod[props.article.slug] = "POST";
            }

            props.onFavoritedArticleClicked(
              props.userInformation.token,
              props.article.slug,
              tempMethod[props.article.slug],
              props.currentPageOffSet,
            );
            props.setHttpMethod(tempMethod);

            // Display add 1 or not
            setFavoritesCountPlus(
              tempMethod[props.article.slug] === "POST" ? 1 : 0
            );
          }
        }}>
        {children}
      </button>
    );
  };

  return (
    <Button>
      <img src='../icon/002-heart-2.png' alt='' />
      {props.article.favoritesCount + favoritesCountPlus}
    </Button>
  );
};


FavoritedButton.propTypes = {
  userInformation: PropTypes.object,
  article: PropTypes.object,
  onFavoritedArticleClicked: PropTypes.func,
  setHttpMethod: PropTypes.func,
};