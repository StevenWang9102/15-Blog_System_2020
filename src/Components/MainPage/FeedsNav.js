import React from "react";
import { Link } from "react-router-dom";
import { displayLimit, offset } from "../../ReduxStore/HttpClient";

export const FeedsNav = props => {
  let isExist;
  let myStyle,
    url,
    navDisplayName = "";

  // Global Feed Nav Config
  if (props.navName === "Global Feed") {
    navDisplayName = "Global Feed";
    isExist = true;
    myStyle = props.globalNav;
    url = "/home/global_feed";
  }

  // Your Feed Nav Config
  if (props.navName === "Your Feed") {
    navDisplayName = "Your Feed";
    isExist = props.userInformation && props.userInformation.token;
    myStyle = props.yourNav;
    url = "/home/your_feed";
  }

  // Popular Feed Nav Config
  if (props.navName === "Popular Feed") {
    navDisplayName = `# ${props.currentTagName}`;
    isExist = props.currentTagName;
    myStyle = props.popularNav;
    url = "/home/popular_tags";
  }

  // My Articles Nav Config
  if (props.navName === "My Articles") {
    navDisplayName = "My Articles";
    isExist = true;
    myStyle = props.profileNavStatusLeft;
    url = `/user_profile/${props.author_name}/my_articles`;
  }

  // Favorited Articles Nav Config
  if (props.navName === "Favorited Articles") {
    navDisplayName = "Favorited Articles";
    isExist = true;
    myStyle = props.profileNavStatusRight;
    url = `/user_profile/${props.author_name}/favorited_articles`;
  }
  return (
    <span>
      {isExist && (
        <Link
          onClick={() => {
            // Your Feed Click
            if (props.navName === "Your Feed") {
              props.setHomeNavStatus("active", "null", "null");
              props.loadYourFeedArticles(displayLimit, offset);
            }

            // Global Feed Click
            if (props.navName === "Global Feed") {
              props.setHomeNavStatus("null", "active", "null");
              props.loadGlobalFeeds(displayLimit, offset);
            }

            // Popular Tag Click
            if (props.navName === "Popular Feed") {
              props.setHomeNavStatus("null", "active", "null");
              props.popularNavClean();
            }

            // My Article Click
            if (props.navName === "My Articles") {
              props.setProfileNavStatus("active", "null");
              props.loadUserProfileDetail(
                props.author_name,
                displayLimit,
                offset
              );
            }

            // Favorited Articles Click
            if (props.navName === "Favorited Articles") {
              props.setProfileNavStatus("null", "active");
              props.onFavoritedArticleNavClicked(
                props.author_name,
                displayLimit,
                offset
              );
            }
          }}
          className={`nav-link display-inline ${myStyle} `}
          to={url}>
          {navDisplayName}
        </Link>
      )}
    </span>
  );
};
