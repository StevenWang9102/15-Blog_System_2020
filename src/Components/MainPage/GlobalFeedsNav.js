import React from "react";
import { Link } from "react-router-dom";
import { displayLimit, offset } from "../../Functions/HttpClient";

export const GlobalFeedsNav = props => {
  return (
    <Link
      // onClick={() => {
      //   props.setLoading("LOADING");
      //   props.setHomeNavStatus("null", "active", "null");
      //   props.loadGlobalFeeds(displayLimit, offset);
      // }}
      // className={`nav-link ${props.favoriteNav} display-inline`}
      to='/home/global_feed'>
      Global Feed
    </Link>
  );
};
