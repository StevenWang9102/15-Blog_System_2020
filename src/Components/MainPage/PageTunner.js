import React, { useState } from "react";
import { displayLimit } from "../../ReduxStore/HttpClient";
import { createUseStyles } from "react-jss";
import { Link } from "react-router-dom";

const useStyles = createUseStyles({
  myButton: {
    flexBasis: "1",
    height: "30px",
    weight: "30px",
    border: "0px",
    padding: "0px",
    borderRadius: "4px",
    margin: "5px",
    backgroundColor: props => props.clickedPage,

    "& a": {
      padding: props => props.padding
    },

    "&:hover": {
      opacity: 0.7,
      backgroundColor: "rgba(0, 0, 0, 0.150)"
    }
  }
});

export const PageTunner = props => {
  const [clickedIndex, setClickedIndex] = useState(-1);

  const Page = ({ children, ...props }) => {
    const classes = useStyles(props);
    return <button className={`${classes.myButton}`}>{children}</button>;
  };

  // Make an Page Number List
  const pageNumber = Math.round(props.articlesAllCount / displayLimit);
  const myPageNumArray = [];
  for (let i = 1; i <= pageNumber; i++) {
    myPageNumArray.push(i);
  }

  return (
    <nav>
      <ul className='pagination'>
        {pageNumber !== 1 &&
          myPageNumArray.map((pageNumber, index) => {
            let offset = index * displayLimit;
            let currentTag = "";
            if (props.globalNav === "active") currentTag = "global_feed";
            if (props.yourNav === "active") currentTag = "your_feed";
            if (props.popularNav === "active") currentTag = "popular_feed";

            // Deal with page padding
            let padding = "";
            if (pageNumber < 10) {
              padding = "10px 15px 10px 15px";
            } else if (pageNumber >= 10 && pageNumber < 20) {
              padding = "5px 13px 5px 13px";
            } else {
              padding = "0px 11px 0px 11px";
            }

            // Deal with choosen page's background
            let clickedPage = "rgba(0, 0, 0, 0.027)";
            if (index === clickedIndex) clickedPage = "rgb(55, 168, 55, 0.2)";

            // Pages on "ArticlePriview" Page
            if (props.fromPage === "ArticlePriview") {
              return (
                <Page padding={padding} key={index} clickedPage={clickedPage}>
                  <Link
                    to={`/home/${currentTag}/p${pageNumber}`}
                    onClick={() => {                      
                      setClickedIndex(index);
                      props.setCurrentPageOffSet(offset);
                      if (currentTag === "global_feed")
                        props.loadGlobalFeeds(displayLimit, offset);
                      else if (currentTag === "your_feed")
                        props.loadYourFeedArticles(displayLimit, offset);
                      else if (currentTag === "popular_feed"){
                        props.onPopularTagClicked(props.currentTagName, displayLimit, offset);
                      }
                    }}>
                    {pageNumber}
                  </Link>
                </Page>
              );
            } else {
              // Pages on "User Profile" Page
              return (
                <Page padding={padding} key={index} clickedPage={clickedPage}>
                  <Link
                    onClick={() => {
                      if (props.profileNavStatusLeft === "active")
                        props.loadUserProfileDetail(
                          props.author_name,
                          displayLimit,
                          offset
                        );
                      else
                        props.onFavoritedArticleNavClicked(
                          props.author_name,
                          displayLimit,
                          offset
                        );
                    }}>
                    {pageNumber}
                  </Link>
                </Page>
              );
            }
          })}
      </ul>
    </nav>
  );
};
