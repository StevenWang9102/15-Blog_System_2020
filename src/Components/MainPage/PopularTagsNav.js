import React from "react";
import { Link } from "react-router-dom";

export const PopularTagsNav = props => {
  return (
    <span>
      {props.currentTagName && (
        <Link
          className={`nav-link ${props.popularNav} display-inline `}
          to='/home/popular_tags'>
          # {props.currentTagName}
        </Link>
      )}
    </span>
  );
};
