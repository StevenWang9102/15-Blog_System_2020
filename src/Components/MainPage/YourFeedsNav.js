import React from "react";
import { Link } from "react-router-dom";
import {
  articleCountDisplay,
  offset
} from "../../Functions/HttpClient";

export const YourFeedsNav = props => {
  
  return (
    <span>
      {props.userInformation && props.userInformation.token && (
        <Link
          onClick={() => {
            props.setLoading("LOADING");
            props.loadYourFeedArticles(articleCountDisplay, offset);
            props.setHomeNavStatus("active", "null", "null");
          }}
          className={`nav-link display-inline ${props.yourNav} `}
          to='/home/your_feed'>
          Your Feed
        </Link>
      )}
    </span>
  );
};
