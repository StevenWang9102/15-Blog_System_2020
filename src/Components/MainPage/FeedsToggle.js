import React from "react";
// import { Link } from "react-router-dom";
// import { displayLimit, offset } from "../../Functions/HttpClient";
import { FeedsNav } from "../../Components/MainPage/FeedsNav";

export const FeedsToggle = props => {
let renderArray =[]
if(props.onPage === "User Profile") renderArray = ["My Articles", "Favorited Articles"];
if(props.onPage === "Article Preview") renderArray = ["Your Feed", "Global Feed", "Popular Feed"];

  return (
    <div className='articles-toggle'>
      <ul className='nav nav-pills outline-active'>
    {
        renderArray.map((feed, index) => {
            props.onPage === "User Profile"? ():()

        }
    }

        {/* {renderArray.map((feed, index) => {
          {props.onPage === "User Profile"? 
          (
            <li className='nav-item'>
              <FeedsNav
                navName={feed}
                author_name={props.author_name}
                setLoading={props.setLoading}
                setProfileNavStatus={props.setProfileNavStatus}
                profileNavStatusLeft={props.profileNavStatusLeft}
                profileNavStatusRight={props.profileNavStatusRight}
                loadUserProfileDetail={props.loadUserProfileDetail}
                onFavoritedArticleNavClicked={props.onFavoritedArticleNavClicked}
              />
            </li>
          ):(<li className='nav-item'>
          <FeedsNav
             navName={feed}
             userInformation={props.userInformation}
             setLoading={props.setLoading}
             loadYourFeedArticles={props.loadYourFeedArticles}
             setHomeNavStatus={props.setHomeNavStatus}
             yourNav={props.yourNav}
             loadGlobalFeeds={props.loadGlobalFeeds}
                globalNav={props.globalNav}
                popularNav={props.popularNav}
                currentTagName={props.currentTagName}
                onFavoritedArticleClicked={props.onFavoritedArticleClicked}/>
        </li>)            
          }         */}
          </ul>
    </div>
  );
};
