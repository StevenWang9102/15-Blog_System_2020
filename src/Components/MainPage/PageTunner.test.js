import React from "react";
import { PageTunner } from "../MainPage/PageTunner";
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const tree = renderer
    .create(
      <PageTunner
        articlesAllCount={0}
        fromPage='test'
        loadUserProfileDetail={() => {}}
        loadGlobalFeeds={() => {}}
        onFavoritedArticleNavClicked={() => {}}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
