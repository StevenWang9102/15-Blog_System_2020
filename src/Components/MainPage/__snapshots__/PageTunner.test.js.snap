// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<nav>
  <ul
    className="pagination"
  />
</nav>
`;
